src/main/java/com/revature/studyforce/notification/config/RestConfig.java

```diff
@@ -0,0 +1,56 @@
+package com.revature.studyforce.notification.config;
+
+import com.revature.studyforce.notification.service.StacktraceSubscriptionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.rest.core.config.RepositoryRestConfiguration;
+import org.springframework.data.rest.webmvc.config.RepositoryRestConfigurer;
+
+import javax.persistence.EntityManager;
+import javax.persistence.metamodel.EntityType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * @author Ronald Lopex
+ */
+@Configuration
+public class RestConfig implements RepositoryRestConfigurer {
+
+    private final EntityManager entityManager;
+
+    @Autowired
+    public RestConfig(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+    @Override
+    public void configureRepositoryRestConfiguration(RepositoryRestConfiguration config) {
+        exposeIds(config);
+    }
+
+    /**
+     * Sets the configuration of the HATOES Links
+     * @param config
+     */
+    private void exposeIds(RepositoryRestConfiguration config) {
+
+        // get a list of all entity classes from teh entity manager
+        Set<EntityType<?>> entities = entityManager.getMetamodel().getEntities();
+
+        // create an array of the entity types
+        List<Class> entityClasses = new ArrayList<>();
+
+        // get the entity types for the entities
+        for (EntityType entityType : entities) {
+            entityClasses.add(entityType.getJavaType());
+        }
+
+        // expose the entity ids for the array of the entity/domain types
+        Class[] domainTypes = entityClasses.toArray(new Class[0]);
+        config.exposeIdsFor(domainTypes);
+
+    }
+
+}
```

Comment:
should be moved to package src/main/java/com/revature/studyforce/configuration
or considered for deletion.

Reply:
deleted